apiserver/facades/agent/uniter/uniter.go

```diff
@@ -124,8 +120,65 @@ func (u *UniterAPI) OpenedMachinePortRangesByEndpoint(args params.Entities) (par
 	return result, nil
 }
 
-// OpenedApplicationPortRangesByEndpoint returns the port ranges opened by each
-// application grouped by application endpoint.
+func (u *UniterAPI) getOneMachineOpenedPortRanges(canAccess common.AuthFunc, machineTag string) (state.MachinePortRanges, error) {
+	tag, err := names.ParseMachineTag(machineTag)
+	if err != nil {
+		return nil, apiservererrors.ErrPerm
+	}
+	if !canAccess(tag) {
+		return nil, apiservererrors.ErrPerm
+	}
+	machine, err := u.getMachine(tag)
+	if err != nil {
+		return nil, err
+	}
+	return machine.OpenedPortRanges()
+}
+
+// OpenedPortRangesByEndpoint returns the port ranges opened by the unit.
+func (u *UniterAPI) OpenedPortRangesByEndpoint() (params.OpenPortRangesByEndpointResults, error) {
+	result := params.OpenPortRangesByEndpointResults{
+		Results: make([]params.OpenPortRangesByEndpointResult, 1),
+	}
+
+	authTag := u.auth.GetAuthTag()
```

Comment:
Aren't we missing a permission check like accessUnit() here?

Reply:
This facade method does not have an arg, we are using the authTag to access its own unit.
So I think we are ok here.